### PR TITLE
Added date_consider_month/year and CI tests

### PR DESCRIPTION
**Pull request by Rethous**
I added the possibility to create all **date_consider/1** for all dates in a given month or a whole year. Also added CI tests for those, to test if _'illegal'_ dates get created.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,15 +4,35 @@
...
+
+Creates all dates of the given year that need to be considered.
+
+|Argument | Domain                     | Note          |
+|:--------|:---------------------------|:--------------|
+| Y       | number (1XXX-2XXX)         | Year
+
 ## date_consider/1( (D,M,Y) )
 
+It is recommended to use date_consider_month/2 or date_consider_year/1, because it prevents the accidental creation of 'illegal' dates i.e. 29th of February in a non leap year.
```
Comment (sthiele) [inline README.md]: Maybe better.
:warning: Do not use!  It allows the creation of  *illegal* dates i.e. 29th of February in a non leap year.
It is recommended to use date_consider_month/2 or date_consider_year/1

File: date-lib.lp [hunk 2]
```diff
@@ -134,6 +134,52 @@ date_relative((D,M,Y),Julian-OJulian) :-
...
+% date_consider given year
+%
+
+date_consider_month(1..12,YYYY) :- date_consider_year(YYYY).
+
+
+%
 %
 % Testcases
 %
```
Comment (sthiele) [inline date-lib.lp]: I think the test cases can now be removed from the encoding.

File: tests/test_date.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import clingo
+import sys
+
+class MyModel:
```
Comment (sthiele) [inline tests/test_date.py]: Is this class needed?
- cost is never used

File: tests/test_date.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    def __init__(self):
+        self.answer = []
+        self.cost = None
+
+def run(instance):
+    ctl = clingo.Control()
+    m = MyModel()
```
Comment (sthiele) [inline tests/test_date.py]: As `answer` is always turned into a set just use a set here.

**Review by sthiele — COMMENTED** [review]
Thanks for the PR :)

File: README.md (hunk 1 above)
Comment (sthiele) [reply README.md]: @MaxOstrowski  Maybe we should remove date_consider/1 from the README at all. If it should not be used as Input predicate. I wish we could make predicates private.

**Comment by sthiele** [comment]
The encoding contains an old duplicate of the documentation. These will be hard to keep in sync. I recommend to remove it.
https://github.com/potassco/asp-date-lib/blob/85f54561d14845763d4f5fc1433c5a75d2553f8a/date-lib.lp#L1-L70

File: tests/test_date.py (hunk 4 above)
Comment (sthiele) [reply tests/test_date.py]:
```
m = {}
def on_model(model):
       m = set(model.symbols(atoms=True))
```

File: date-lib.lp [hunk 5]
```diff
@@ -134,6 +134,52 @@ date_relative((D,M,Y),Julian-OJulian) :-
...
+last_date_month(28,2,YYYY) :- 
+	date_consider_month(2,YYYY), 
+	not is_leap_year(YYYY).
+
+last_date_month(29,2,YYYY) :- 
+	date_consider_month(2,YYYY), 
+	is_leap_year(YYYY).
+
+%
+% date_consider given month
```
Comment (sthiele) [inline date-lib.lp]: Improve comment.

File: date-lib.lp [hunk 6]
```diff
@@ -134,6 +134,52 @@ date_relative((D,M,Y),Julian-OJulian) :-
...
+%
+
+date_consider((1..DD,MM,YYYY)) :-
+	date_consider_month(MM,YYYY),	
+	last_date_month(DD,MM,YYYY).
+
+%
+% date_consider given year
```
Comment (sthiele) [inline date-lib.lp]: Improve comment.

File: README.md (hunk 1 above)
Comment (sthiele) [reply README.md]: Maybe add integrity constraints for *illegal* dates? or a save alternative like `checked_date_consider`?

**Comment by MaxOstrowski** [comment]
Thanks for your comments @sthiele.
The plan is to mark `date_consider` as deprecated and to remove it from the readme otherwise.

File: README.md [hunk 7]
```diff
@@ -4,14 +4,35 @@
...
+Creates all dates of the given year that need to be considered.
+
+|Argument | Domain                     | Note          |
+|:--------|:---------------------------|:--------------|
+| Y       | number (1XXX-2XXX)         | Year
+
+## (deprecated) date_consider/1( (D,M,Y) )
+
+Do not use! It allows the creation of illegal dates i.e. 29th of February in a non leap year.
+It is recommended to use date_consider_month/2 or date_consider_year/1
```
Comment (MaxOstrowski) [inline README.md]: missing dot

**Comment by MaxOstrowski** [comment]
last_date_month is not documented as an output predicate ?

**Comment by Rethous** [comment]
I believe I have addressed all comments and made the proper changes.
About marking date_consider/1 as deprecated. I was think about moving it to the output predicates, since that would describe its current role more accurately. 
What do you think? @sthiele @MaxOstrowski

**Comment by MaxOstrowski** [comment]
Actually sounds like a very good idea. I actually needed that already.
Do add the description as output predicate and I will merge.
Thanks a lot!

**Pull request merged**